agent/eni/watcher/watcher_test.go

```diff
@@ -0,0 +1,283 @@
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package watcher
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	apieni "github.com/aws/amazon-ecs-agent/agent/api/eni"
+	mock_dockerstate "github.com/aws/amazon-ecs-agent/agent/engine/dockerstate/mocks"
+	"github.com/aws/amazon-ecs-agent/agent/statechange"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	expirationTimeAddition    = time.Second * 5
+	expirationTimeSubtraction = time.Second * -1
+
+	primaryMAC = "02:22:ea:8c:80:ae"
+	randomMAC  = "00:0a:95:9d:68:16"
+)
+
+// Test for SendENIStateChange. We send a StateChange notification for an ECS managed ENI which is yet to expire.
+func TestSendENIStateChange(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockStateManager := mock_dockerstate.NewMockTaskEngineState(mockCtrl)
+	eventChannel := make(chan statechange.Event)
+	ctx := context.TODO()
+
+	gomock.InOrder(
+		mockStateManager.EXPECT().ENIByMac(randomMAC).Return(&apieni.ENIAttachment{
+			ExpiresAt: time.Now().Add(expirationTimeAddition),
+		}, true),
+	)
+
+	watcher := &ENIWatcher{
```

Comment:
As you are testing the watcher under different conditions, can you create a helper function `setupWatcher` to help setup the test infrastructure and then invoke it to obtain the watcher in the tests.